core/src/main/java/org/opensearch/sql/utils/DateTimeUtils.java

```diff
@@ -139,4 +139,15 @@ public static LocalDateTime extractDateTime(ExprValue value,
         ? ((ExprTimeValue) value).datetimeValue(functionProperties)
         : value.datetimeValue();
   }
+
+  /**
+   * Extracts Instant from a datetime ExprValue.
+   * Uses `FunctionProperties` for `ExprTimeValue`.
+   */
+  public static Instant extractTimestamp(ExprValue value,
+                                         FunctionProperties functionProperties) {
+    return value instanceof ExprTimeValue
+        ? ((ExprTimeValue) value).timestampValue(functionProperties)
+        : value.timestampValue();
```

Comment:
Oooh.   I do not like this logic is required here.  
It would be better if the logic was in Expr*Value classes.

Reply:
Can you give an example, please?